src/ExpressionParser/Infix/BinaryOperatorExpressionParser.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\ExpressionParser\Infix;
+
+use Twig\ExpressionParser\AbstractExpressionParser;
+use Twig\ExpressionParser\InfixAssociativity;
+use Twig\ExpressionParser\InfixExpressionParserInterface;
+use Twig\ExpressionParser\PrecedenceChange;
+use Twig\Node\Expression\AbstractExpression;
+use Twig\Parser;
+use Twig\Token;
+
+class BinaryOperatorExpressionParser extends AbstractExpressionParser implements InfixExpressionParserInterface
+{
+    public function __construct(
+        /** @var class-string<AbstractExpression> */
```

Comment:
should it support any AbstractExpression or should it enforce `AbstractBinary` as it expects the constructor signature of this class ? And should we make `AbstractBinary`  implement an (internal) interface enforcing the constructor signature ?

Reply:
Done